client/app/components/collection-item.js

```diff
@@ -1,6 +1,16 @@
 import Ember from 'ember';
 
 export default Ember.Component.extend({
+    item: null,
+    isGroup: Ember.computed(function() {
+       var type = this.get('item').get('type');
+       return !type;
+    }),
+    type: Ember.computed(function() {
+        var item = this.get('item');
+        var type = item.get('type');
+        return type ? type : 'group'
```

Comment:
This doesn't feel like very robust to me because type definition may just not be there. Is the best way to deal with undefined types to default them to group or show them as errors?